src/Datagrid/Pager.php

```diff
@@ -45,7 +45,7 @@ public function computeNbResult()
             current($this->getCountColumn())
         ));
 
-        return $countQuery->resetDQLPart('orderBy')->getQuery()->getSingleScalarResult();
+        return (int) ($countQuery->resetDQLPart('orderBy')->getQuery()->getSingleScalarResult());
```

Comment:
Technically a BC-break, but I think people actually convert this themselves to int if they figured out this was a string, or use implicit conversion.